core/capabilities/syncer_test.go

```diff
@@ -32,9 +38,565 @@ func TestSyncer_CleanStartClose(t *testing.T) {
+					IsPublic:         true,
+					AcceptsWorkflows: true,
+					NodeP2PIds:       workflowDonNodes,
+				},
+				{
+					Id:               capDonID,
+					ConfigCount:      uint32(0),
+					F:                uint8(1),
+					IsPublic:         true,
+					AcceptsWorkflows: false,
+					NodeP2PIds:       capabilityDonNodes,
+					CapabilityConfigurations: []kcr.CapabilityRegistryCapabilityConfiguration{
+						{
+							CapabilityId: triggerCapID,
+							Config:       []byte(""),
+						},
+						{
+							CapabilityId: targetCapID,
+							Config:       []byte(""),
+						},
+					},
+				},
+			},
+			IDsToCapabilities: map[capabilityID]kcr.CapabilityRegistryCapability{
+				triggerCapID: {
+					LabelledName:   "streams-trigger",
+					Version:        "1.0.0",
+					CapabilityType: 0,
+				},
+				targetCapID: {
+					LabelledName:   "write-chain_evm_1",
+					Version:        "1.0.0",
+					CapabilityType: 3,
+				},
+			},
+			IDsToNodes: map[p2ptypes.PeerID]kcr.CapabilityRegistryNodeInfo{
+				capabilityDonNodes[0]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               capabilityDonNodes[0],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				capabilityDonNodes[1]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               capabilityDonNodes[1],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				capabilityDonNodes[2]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               capabilityDonNodes[2],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				capabilityDonNodes[3]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               capabilityDonNodes[3],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				workflowDonNodes[0]: {
+					NodeOperatorId: 1,
+					Signer:         randomWord(),
+					P2pId:          workflowDonNodes[0],
+				},
+				workflowDonNodes[1]: {
+					NodeOperatorId: 1,
+					Signer:         randomWord(),
+					P2pId:          workflowDonNodes[1],
+				},
+				workflowDonNodes[2]: {
+					NodeOperatorId: 1,
+					Signer:         randomWord(),
+					P2pId:          workflowDonNodes[2],
+				},
+				workflowDonNodes[3]: {
+					NodeOperatorId: 1,
+					Signer:         randomWord(),
+					P2pId:          workflowDonNodes[3],
+				},
+			},
+		},
+	}
+	syncer := newRegistrySyncer(make(services.StopChan), wrapper, registry, dispatcher, lggr, HardcodedDonNetworkSetup{}, mr)
+	require.NoError(t, err)
+
+	dispatcher.On("SetReceiver", fullTriggerCapID, fmt.Sprint(capDonID), mock.AnythingOfType("*remote.triggerSubscriber")).Return(nil)
+	dispatcher.On("SetReceiver", fullTargetID, fmt.Sprint(capDonID), mock.AnythingOfType("*target.client")).Return(nil)
+
+	err = syncer.sync(ctx)
+	require.NoError(t, err)
+
+	_, err = registry.Get(ctx, fullTriggerCapID)
```

Comment:
Maybe this is where `CapabilityExists` check would come in handy too?

Reply:
I think this is a difference of philosophy :/ I view this as the canonical way to do a `CapabilityExists` check so I wouldn't personally add a CapabilityExists(id) -> bool helper since that would just make the API surface bigger.